src/vivarium_census_prl_synth_pop/utilities.py

```diff
@@ -376,3 +376,23 @@ def update_address_id_for_unit_and_sims(
         pop.loc[rows_changing_address_id_idx, address_id_col_name] = updated_address_ids
 
     return pop, moving_units, total_address_id_count
+
+
+def add_guardian_address_ids(pop: pd.DataFrame) -> pd.DataFrame:
+    """Map the address ids of guardians to each simulant's guardian address columns"""
+    for i in [1, 2]:
+        s_guardian_id = pop[f"guardian_{i}"].dropna()
+        s_guardian_id = s_guardian_id[
+            s_guardian_id != -1
+        ]  # is it faster to remove the negative values?
+        pop[f"guardian_{i}_address_id"] = s_guardian_id.map(pop["address_id"])
+    return pop
+
+
+def convert_middle_name_to_initial(pop: pd.DataFrame) -> pd.DataFrame:
+    """Converts middle names to middle initials. Note that this drops
+    the 'middle_name' column altogether and replaces it with 'middle_initial'
+    """
+    pop["middle_initial"] = pop["middle_name"].str[0]
```

Comment:
I confirmed that everyone does, but this approach also works w/ nans (which is why I used this instead of `.astype(str)[0]`)

```
>>> df = pd.DataFrame({'x':["David", np.nan]})
>>> df.x.str[0]
0      D
1    NaN
Name: x, dtype: object
>>> 
```